### ISSUE TITLE
Error: parse error in "elasticsearch/templates/data-statefulset.yaml": template: elasticsearch/templates/data-statefulset.yaml:106: function "tpl" not defined

### ISSUE THREAD

**Issue by tanermelikoglu** [helm/charts#11599]
** BUG REPORT ** 

**Version of Helm and Kubernetes**:
Client: &version.Version{SemVer:"v2.5.1", 
Server: &version.Version{SemVer:"v2.1.3"

Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3 ...
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.5" ...

**Which chart**:
stable/elasticsearch

**What happened**:
Want to deploy elasticsearch via following helm command:

`helm install stable/elasticsearch --name ${ELASTICSEARCH_NAME} --values elk/elasticsearch-values.yaml --namespace ${NAMESPACE} --set data.storageClass=${STORAGE_CLASS},data.storage=${STORAGE_SIZE}`

The elk/elasticsearch-values.yaml is the same like values.yaml in https://github.com/helm/charts/blob/master/stable/elasticsearch/values.yaml
=>
`Error: parse error in "elasticsearch/templates/data-statefulset.yaml": template: elasticsearch/templates/data-statefulset.yaml:106: function "tpl" not defined`

**Comment by micw** [comment]
You need to update your tiller version (helm init --upgrade). Tiller is the serverside part of helm and should be updated from time to time.

**Comment by juan131** [comment]
Hi @micw 

From you last comment, I understand you solved the issue by upgrading the Tiller version. Could you confirm that and close the issue in that case?

**Comment by tanermelikoglu** [comment]
Hi @micw and @juan131,

thx for your replay.

Nope, it does not work.
1-  `helm update`
2-  
```
helm init --upgrade
$HELM_HOME has been configured at /Users/xyz/.helm.

Tiller (the helm server side component) has been upgraded to the current version.
Happy Helming!
```
```
helm version
Client: &version.Version{SemVer:"v2.5.1", GitCommit:"7cf31e8d9a026287041bae077b09165be247ae66", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.5.1", GitCommit:"7cf31e8d9a026287041bae077b09165be247ae66", GitTreeState:"clean"}
```
3-
```
helm install stable/elasticsearch  --set data.storageClass=mystorage,data.storage=5Gi
Error: parse error in "elasticsearch/templates/configmap.yaml": template: elasticsearch/templates/configmap.yaml:1: function "regexFind" not defined
```

**Comment by micw** [comment]
@juan131 As you can see, the helm version you installed supports the "tpl" function, so the original issue is solved. Seems that there's another function "regexFind" that your helm version does not support. You are using 2.5.x, current is 2.12.x. I'm pretty sure that the new issue will also be gone after you installed to a recent helm version.

@tanermelikoglu I did not test this, just tried to help because I saw that the issue was/is almost certain caused by the outdated helm version the reporter is using.

**Comment by tanermelikoglu** [comment]
@micw Fixed throught upgrading of Helm version v2.13.0 (client and server).
thx.
